src/Behat/Behat/Output/Node/Printer/ListPrinter.php

```diff
@@ -45,24 +46,26 @@ final class ListPrinter
      * @var TranslatorInterface
      */
     private $translator;
-    /**
-     * @var string
-     */
-    private $basePath;
+
+    private ConfigurablePathPrinter $configurablePathPrinter;
 
     /**
      * @param ExceptionPresenter $exceptionPresenter deprecated , will be removed in the next major version
-     * @param string $basePath
+     * @param string $basePath deprecated , will be removed in the next major version
      */
     public function __construct(
         ResultToStringConverter $resultConverter,
         ExceptionPresenter $exceptionPresenter,
         TranslatorInterface $translator,
-        $basePath,
+        string $basePath,
     ) {
         $this->resultConverter = $resultConverter;
         $this->translator = $translator;
-        $this->basePath = $basePath;
+    }
+
+    public function setConfigurablePathPrinter(ConfigurablePathPrinter $configurablePathPrinter)
```

Comment:
We add a function to be able to set the new configurable path printer service as we cannot inject it through the constructor to keep BC